src/fondant/pipeline/runner.py

```diff
@@ -66,22 +67,62 @@ def run(
             to mount in the docker-compose spec.
             build_args: List of build arguments to pass to docker
         """
-        if isinstance(input, Pipeline):
-            os.makedirs(".fondant", exist_ok=True)
-            output_path = ".fondant/compose.yaml"
-            logging.info(
-                "Found reference to un-compiled pipeline... compiling",
+        if self.docker_is_installed() and self.docker_compose_is_installed():
```

Comment:
This is a bit of a strange construct with an `if` and no `else`. Reading this, I would expect us to just do nothing if `docker` or `docker compose` is not installed, while I see that we actually raise an error in those methods.

So I would either move the error messages into an `else` block here, or don't use them in an `if` block (and rename them):
```python
self.check_docker_install()
self.check_docker_compose_install()

...
```

Reply:
```
try:
   self.check_docker_install()
   self.check_docker_compose_install()
except:
    ?
```
no ?

Reply:
I've thought it adds some readability. I removed the if block and renamed the functions.